tfx/experimental/pipeline_testing/examples/chicago_taxi_pipeline/taxi_pipeline_regression_e2e_test.py

```diff
@@ -0,0 +1,131 @@
+    # Metadata path for recording successful pipeline run.
+    self._metadata_db_uri = os.path.join(self._test_dir, 'tfx', 'record',
+                                         'metadata.db')
+    # Metadata path for stub pipeline runs.
+    self._metadata_path = os.path.join(self._test_dir, 'tfx', 'metadata',
+                                       self._pipeline_name, 'metadata.db')
+    self._output_dir = os.path.join(self._test_dir, 'testdata')
+
+  def testTaxiPipelineBeam(self):
+    # Runs the pipeline and record to self._output_dir
+    record_taxi_pipeline = taxi_pipeline_beam._create_pipeline(  # pylint:disable=protected-access, unexpected-keyword-arg
+        pipeline_name=self._pipeline_name,
+        data_root=self._data_root,
+        module_file=self._module_file,
+        serving_model_dir=self._serving_model_dir,
+        pipeline_root=self._pipeline_root,
+        metadata_path=self._metadata_db_uri,
+        beam_pipeline_args=[])
+    BeamDagRunner().run(record_taxi_pipeline)
+    pipeline_recorder_utils.record_pipeline(
+        output_dir=self._output_dir,
+        metadata_db_uri=self._metadata_db_uri,
+        host=None,
+        port=None,
+        pipeline_name=self._pipeline_name,
+        run_id=None)
+
+    # Run pipeline with stub executors first time.
+    taxi_pipeline = taxi_pipeline_beam._create_pipeline(  # pylint:disable=protected-access, unexpected-keyword-arg
+        pipeline_name=self._pipeline_name,
+        data_root=self._data_root,
+        module_file=self._module_file,
+        serving_model_dir=self._serving_model_dir,
+        pipeline_root=self._pipeline_root,
+        metadata_path=self._metadata_path,
+        beam_pipeline_args=[])
+
+    model_resolver_id = 'ResolverNode.latest_blessed_model_resolver'
+    self._component_ids = [component.id
+                           for component in taxi_pipeline.components
+                           if component.id != model_resolver_id]
+
+    my_launcher = stub_component_launcher.get_stub_launcher_class(
+        test_data_dir=self._output_dir,
+        stubbed_component_ids=self._component_ids,
+        stubbed_component_map={})
+    my_pipeline_config = pipeline_config.PipelineConfig(
+        supported_launcher_classes=[
+            my_launcher,
+        ])
+    BeamDagRunner(config=my_pipeline_config).run(taxi_pipeline)
+
+    self.assertTrue(tf.io.gfile.exists(self._metadata_path))
+    metadata_config = metadata.sqlite_metadata_connection_config(
+        self._metadata_path)
+    with metadata.Metadata(metadata_config) as m:
+      artifact_count = len(m.store.get_artifacts())
+      execution_count = len(m.store.get_executions())
+      self.assertGreaterEqual(artifact_count, execution_count)
+      self.assertLen(taxi_pipeline.components, execution_count)
+
+    # Run pipeline with stub executors second time.
+    BeamDagRunner(config=my_pipeline_config).run(taxi_pipeline)
+
+    # All executions but Evaluator and Pusher are cached.
```

Comment:
Is this what we want to test? I think that cached execution is not what we really want...

Reply:
changed to testing whether the recorded files are successfully copied to output uris.